src/server/game/Server/Packets/CalendarPackets.cpp

```diff
@@ -0,0 +1,460 @@
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "CalendarPackets.h"
+
+ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::Calendar::CalendarSendCalendarEventInfo const& eventInfo)
+{
+    data << uint64(eventInfo.EventID);
+    data << uint8(eventInfo.EventType);
+    data.AppendPackedTime(eventInfo.Date);
+    data << uint32(eventInfo.Flags);
+    data << int32(eventInfo.TextureID);
+    data << eventInfo.EventGuildID;
+    data << eventInfo.OwnerGuid;
+
+    data.WriteBits(eventInfo.EventName.size(), 8);
+    data.FlushBits();
+    data.WriteString(eventInfo.EventName);
+
+    return data;
+}
+
+ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::Calendar::CalendarSendCalendarRaidResetInfo const& resetInfo)
+{
+    data << int32(resetInfo.MapID);
+    data << uint32(resetInfo.Duration);
+    data << int32(resetInfo.Offset);
+
+    return data;
+}
+
+ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::Calendar::CalendarSendCalendarRaidLockoutInfo const& lockoutInfo)
+{
+    data << uint64(lockoutInfo.InstanceID);
+    data << int32(lockoutInfo.MapID);
+    data << uint32(lockoutInfo.DifficultyID);
+    data << uint32(lockoutInfo.ExpireTime);
+
+    return data;
+}
+
+ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::Calendar::CalendarSendCalendarInviteInfo const& inviteInfo)
+{
+    data << uint64(inviteInfo.EventID);
+    data << uint64(inviteInfo.InviteID);
+    data << uint8(inviteInfo.Status);
+    data << uint8(inviteInfo.Moderator);
+    data << uint8(inviteInfo.InviteType);
+    data << inviteInfo.InviterGuid;
+
+    return data;
+}
+
+ByteBuffer& operator<<(ByteBuffer& data, WorldPackets::Calendar::CalendarEventInviteInfo const& inviteInfo)
+{
+    data << inviteInfo.Guid;
+    data << uint64(inviteInfo.InviteID);
+
+    data << uint8(inviteInfo.Level);
+    data << uint8(inviteInfo.Status);
+    data << uint8(inviteInfo.Moderator);
+    data << uint8(inviteInfo.InviteType);
+
+    data.AppendPackedTime(inviteInfo.ResponseTime);
+
+    data.WriteBits(inviteInfo.Notes.size(), 8);
+    data.FlushBits();
+    data.WriteString(inviteInfo.Notes);
+
+    return data;
+}
+
+void WorldPackets::Calendar::CalendarGetEvent::Read()
+{
+    _worldPacket >> EventID;
+}
+
+void WorldPackets::Calendar::CalendarGuildFilter::Read()
+{
+    _worldPacket >> MinLevel;
+    _worldPacket >> MaxLevel;
+    _worldPacket >> MaxRankOrder;
+}
+
+void WorldPackets::Calendar::CalendarAddEvent::Read()
+{
+    _worldPacket.FlushBits();
+    uint8 titleLength = _worldPacket.ReadBits(8);
+    uint16 descriptionLength = _worldPacket.ReadBits(11);
+
+    _worldPacket >> EventInfo.EventType;
+    _worldPacket >> EventInfo.TextureID;
+    EventInfo.Time = _worldPacket.ReadPackedTime();
+    _worldPacket >> EventInfo.Flags;
+
+    EventInfo.Invites.resize(_worldPacket.read<int32>());
```

Comment:
You must never blindly trust what players send when it comes to sizes like this.